mne/time_frequency/psd.py

```diff
@@ -112,6 +114,202 @@ def _check_nfft(n, n_fft, n_overlap):
     return n_fft, n_overlap
 
 
+def _check_psd_data(inst, tmin, tmax, picks, proj):
+    """Helper to do checks on PSD data / pull arrays from inst"""
+    from ..io.base import _BaseRaw
+    from ..epochs import _BaseEpochs
+    from ..evoked import Evoked
+    if not isinstance(inst, (_BaseEpochs, _BaseRaw, Evoked)):
+        raise ValueError('epochs must be an instance of Epochs, Raw, or'
+                         'Evoked. Got type {0}'.format(type(inst)))
+
+    time_mask = _time_mask(inst.times, tmin, tmax)
+    if picks is None:
+        picks = pick_types(inst.info, meg=True, eeg=True, ref_meg=False,
+                           exclude='bads')
+    if proj:
+        inst = inst.apply_proj()
```

Comment:
Actually maybe you should just do:
```
inst = inst.copy().apply_proj()
```
It's easy enough, and if we run into memory problems we can refactor later.